packages/forklift-console-plugin/src/modules/Providers/utils/components/CertificateUpload/CertificateUpload.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC } from 'react';
+import { useModal } from 'src/modules/Providers/modals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { Button, FileUpload, FileUploadProps, Flex, FlexItem } from '@patternfly/react-core';
+
+import { FetchCertificateModal } from './FetchCertificateModal';
+
+export interface CertificateUploadProps {
+  id?: string;
+  url: string;
+  cacert: string;
+  validated: FileUploadProps['validated'];
+  isDisabled: boolean;
+  handleSave: (cacert: string) => void;
+}
+
+/**
+ * Provide the certificate using following paths:
+ * 1. manual copy-paste (provided by FileUpload widget)
+ * 2. uploading a file (provided by FileUpload widget)
+ * 3. fetch from the specified URL (via tls-certificate endpoint) end verify
+ */
+export const CertificateUpload: FC<CertificateUploadProps> = ({
+  id,
+  url,
+  cacert,
+  handleSave,
+  validated,
+  isDisabled,
+}) => {
+  const { showModal } = useModal();
+  const { t } = useForkliftTranslation();
+  return (
+    <>
+      <FileUpload
+        id={id}
+        type="text"
+        filenamePlaceholder={t('Drag and drop a file or upload one')}
+        value={cacert}
+        validated={validated}
+        onDataChange={handleSave}
+        onTextChange={handleSave}
+        onClearClick={() => handleSave('')}
+        browseButtonText={t('Upload')}
+        isDisabled={isDisabled}
+      >
+        <Flex>
+          <FlexItem>
+            <Button
+              isDisabled={!url || isDisabled}
+              variant="link"
+              onClick={() =>
+                showModal(
+                  <FetchCertificateModal url={url} handleSave={handleSave} existingCert={cacert} />,
```

Comment:
```suggestion
                  <FetchCertificateModal url={url} handleSave={handleSave} />,
```

Reply:
the modal don't need to know about current text, it will override it anyways

Reply:
Not verifying certificate mismatch (and silently overwriting cert content) is a security issue.

Reply:
why do you think it is a security risk ?
what are the scenario you envision ?

Reply:
Already answered about this one to @ahadas  - see [here](https://github.com/kubev2v/forklift-console-plugin/pull/924#issuecomment-1945943374)
The key point is that certs can change in time. If such change is detected the user should be informed about this case and allowed to verify if this is not an attack.

Reply:
> User can provide the cert from file and use the new modal to verify.

the modal is not used for "verify" an already existing certification.

it is used to fetch a cert and then the user has an option to verify the fingerprint manually.

a user verify the fingerprint, if the fingerprint is OK, the cert is OK - if the user only verify a correct fingerprint the certification is safe, regardless if it was changed or not.

Note:
we may need a clearer text here, make sure the user understand - 
a. the link will fetch new certification, not verify the existing one.
b. the user will need to verify that the fetched cert is valid by checking the fingerprint, and only approve cert he can verify using the fingerprint shown.

for example when using `ssh`  on a new host, the ssh utility will ask to verify the fingerprint - meaning verify the new fingerint